client/basic-authorship/src/basic_authorship.rs

```diff
@@ -358,6 +358,10 @@ impl<A, B, Block, C, PR> Proposer<B, Block, C, A, PR>
 
 		self.transaction_pool.remove_invalid(&unqueue_invalid);
 
+		if self.transaction_pool.status().is_empty() {
```

Comment:
This is not correct.

Reply:
It would be great if you can share more, as I have fairly less knowledge about the tx pool and consensus.
Also the behaviour after applied above changes does seem works as expected, 
* if no extrinsics, no blocks generated, 
* after make a balance transfer, a new block authored. 

A potential issue probably is with BABE, this way will result in unexpected epoch changes.

Reply:
The point is that you are modifying the wrong place. 

The issue is speaking about skipping empty blocks for `Aura` and your modification here would skip empty blocks for every consensus engine and that is a bad idea.

It is also not discussed on what the security implications are from requested feature.

Reply:
I see, thanks for pointing it out.